marquez_airflow/dag.py

```diff
@@ -193,6 +206,62 @@ def handle_callback(self, *args, **kwargs):
 
         return super().handle_callback(*args, **kwargs)
 
+    def _alternate_create_flow(self, step, ti):
+        inputs = None
+        if step.inputs:
+            self.register_datasets(step.inputs)
+            inputs = self._to_dataset_ids(step.inputs)
+        outputs = None
+        if step.outputs:
+            self.register_datasets(step.outputs)
+            outputs = self._to_dataset_ids(step.outputs)
+
+        ti_location = self._get_location(ti.task)
+        # Create a job to ensure it exists
+        self.get_or_create_marquez_client().create_job(
+            namespace_name=self.marquez_namespace,
+            job_name=step.name,
+            job_type=JobType.BATCH,
+            location=step.location or ti_location,
+            input_dataset=inputs,
+            output_dataset=outputs,
+            description=self.description)
+        # Run must be called after job creation
+        run_args = {}
+        run_id = self.create_run(step.name, run_args,
+                                 self._to_iso_8601(ti.start_date),
+                                 self._to_iso_8601(ti.end_date))
+        self.get_or_create_marquez_client() \
+            .mark_job_run_as_started(run_id, self._to_iso_8601(ti.start_date))
+
+        # Associate existing output to run
+        if step.outputs:
+            self.register_datasets(step.outputs,
+                                   run_id)
+            self.log.debug(
+                f're-register outputs: {outputs} '
+            )
+
+        # Recreate job to associate run
+        self.get_or_create_marquez_client().create_job(
+            namespace_name=self.marquez_namespace,
+            job_name=step.name,
+            job_type=JobType.BATCH,
+            location=step.location or ti_location,
+            input_dataset=inputs,
+            output_dataset=outputs,
+            description=self.description,
+            run_id=run_id)
+        self._handle_task_instance_state(ti, run_id)
+        pass
+
+    def _get_run_ids(self, ti, run_id, session):
+        job_name = self._marquez_job_name_from_ti(ti)
+
+        marquez_job_run_ids = self._job_id_mapping.pop(
+            job_name, run_id, session)
```

Comment:
I haven't read closely enough to make sure this is what you intended. I'm assuming it is but just to be safe, I'm pointing out that when using this method it also removes the marquez run id. There was a call at some point that needed it to still be there, and just after we called  `_handle_task_instance_state` we don't need them around anymore. We were discussion using a name based UUID derived from job_name and run_id instead to avoid this altogether.

Reply:
I have this fixed in main but it was a bad merge artifact. Good catch.